ThunderCloud/Array+Extensions.swift

```diff
@@ -0,0 +1,25 @@
+//
+//  Array+Extensions.swift
+//  ThunderCloud
+//
+//  Created by Ben Shutt on 03/01/2020.
+//  Copyright © 2020 threesidedcube. All rights reserved.
+//
+
+import Foundation
+
+extension Array {
+    
+    mutating func removeAllAndReturn(where block: (Element) throws -> Bool) rethrows -> [Element] {
```

Comment:
Might be worth renaming to `removingAll(where:` but happy to leave and merge :)

Reply:
I'll update as I'm in agreement with you there, good spot! :D